File: TelerikUIExamples/TelerikUIExamples/AlertViewAnimations.m
//
//  AlertViewAnimations.m
//  TelerikUIExamples
//
//  Copyright (c) 2015 Telerik. All rights reserved.
//

#import <TelerikUI/TelerikUI.h>
#import "UIButton_Circle.h"
#import "AlertViewGettingStarted.h"
#import "AlertViewAnimations.h"

@interface AlertViewAnimations () <TKListViewDelegate>
@end

@implementation AlertViewAnimations
{
    UILabel *_appearLabel;
    UILabel *_dismissLabel;
    TKDataSource *_appearAnimations;
    TKDataSource *_hideAnimations;
    TKListView *_appearAnimationsList;
    TKListView *_hideAnimationsList;
    TKAlert *_alert;
}

- (instancetype)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        [self addOption:@"Show Alert" selector:@selector(show)];
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    _alert = [TKAlert new];
    _alert.title = @"Animations";
    _alert.style.backgroundStyle = TKAlertBackgroundStyleBlur;
    
    [_alert addActionWithTitle:@"Close" handler:^BOOL(TKAlert *alert, TKAlertAction *action) {
        return YES;
    }];

    _appearAnimations = [[TKDataSource alloc] initWithArray:@[@"Scale in", @"Fade in", @"Slide from left", @"Slide from top", @"Slide from right", @"Slide from bottom"]];
    _hideAnimations = [[TKDataSource alloc] initWithArray:@[@"Scale out", @"Fade out", @"Slide to left", @"Slide to top", @"Slide to right", @"Slide to bottom"]];
    
    TKDataSourceListViewSettings_InitCellWithItemBlock block = ^(TKListView *listView, NSIndexPath *indexPath, TKListViewCell *cell, id item) {
        cell.textLabel.font = [UIFont systemFontOfSize:12];
        cell.textLabel.textAlignment = NSTextAlignmentCenter;
        cell.textLabel.text = item;
    };
    
    [_appearAnimations.settings.listView initCell:block];
    [_hideAnimations.settings.listView initCell:block];
    
    [self.view addSubview:[TKListView new]];
    
    _appearAnimationsList = [[TKListView alloc] initWithFrame:CGRectMake(0, 0, self.view.frame.size.width/2, self.view.frame.size.height)];
    _appearAnimationsList.dataSource = _appearAnimations;
    _appearAnimationsList.delegate = self;
    _appearAnimationsList.tag = 0;
    _appearAnimationsList.autoresizingMask = UIViewAutoresizingNone;
    [self.view addSubview:_appearAnimationsList];
    
    [_appearAnimationsList selectItemAtIndexPath:[NSIndexPath indexPathForItem:3 inSection:0] animated:NO scrollPosition:UICollectionViewScrollPositionNone];

    _hideAnimationsList = [[TKListView alloc] initWithFrame:CGRectMake(self.view.frame.size.width/2., 0, self.view.frame.size.width/2., self.view.frame.size.height)];
    _hideAnimationsList.dataSource = _hideAnimations;
    _hideAnimationsList.delegate = self;
    _hideAnimationsList.tag = 1;
    _hideAnimationsList.autoresizingMask = UIViewAutoresizingNone;
    [self.view addSubview:_hideAnimationsList];
    
    [_hideAnimationsList selectItemAtIndexPath:[NSIndexPath indexPathForItem:5 inSection:0] animated:NO scrollPosition:UICollectionViewScrollPositionNone];
    
    _appearLabel = [UILabel new];
    _appearLabel.text = @"Show animation:";
    _appearLabel.textColor = [UIColor blackColor];
    _appearLabel.font = [UIFont systemFontOfSize:12];
    _appearLabel.textAlignment = NSTextAlignmentCenter;
    [self.view addSubview:_appearLabel];
    
    _dismissLabel = [UILabel new];
    _dismissLabel.text = @"Dismiss animation:";
    _dismissLabel.textColor = [UIColor blackColor];
    _dismissLabel.font = [UIFont systemFontOfSize:12];
    _dismissLabel.textAlignment = NSTextAlignmentCenter;
    [self.view addSubview:_dismissLabel];
}

- (void)viewDidLayoutSubviews
{
    CGFloat titleHeight = 60;
    if (UIInterfaceOrientationIsLandscape([UIApplication sharedApplication].statusBarOrientation)) {
        titleHeight = 50;
    }
    
    if ([UIDevice currentDevice].userInterfaceIdiom == UIUserInterfaceIdiomPad) {
        titleHeight += 40;
    }
    
    CGFloat halfWidth = CGRectGetWidth(self.view.frame)/2;
    _appearLabel.frame = CGRectMake(0, titleHeight + 10, halfWidth, 20);
    _dismissLabel.frame = CGRectMake(halfWidth, titleHeight + 10, halfWidth, 20);
    
    CGFloat height = CGRectGetHeight(self.view.frame) - titleHeight - 50;
    
    _appearAnimationsList.frame = CGRectMake(0, titleHeight + 40, halfWidth, height);
    _hideAnimationsList.frame = CGRectMake(halfWidth, titleHeight + 40, halfWidth, height);
}

- (void)show
{
    NSMutableString *message = [NSMutableString new];
    
    NSArray *selected = [_appearAnimationsList indexPathsForSelectedItems];
    if (selected.count > 0) {
        NSIndexPath *indexPath = selected[0];
        [message appendString:[NSString stringWithFormat:@"Alert did %@.\n", [(NSString*)_appearAnimations.items[indexPath.row] lowercaseString]]];
    }

    selected = [_hideAnimationsList indexPathsForSelectedItems];
    if (selected.count > 0) {
        NSIndexPath *indexPath = selected[0];
        [message appendString:[NSString stringWithFormat:@"It will %@ when closed.", [(NSString*)_hideAnimations.items[indexPath.row] lowercaseString]]];
    }
    
    _alert.message = message;
    
    [_alert show:YES];
}

#pragma mark - TKListViewDelegate

- (void)listView:(TKListView *)listView didSelectItemAtIndexPath:(NSIndexPath *)indexPath
{
    if (listView.tag == 0) {
        _alert.style.showAnimation = indexPath.row;
    }
    else {
        _alert.style.dismissAnimation = indexPath.row;
    }
}

@end

File: TelerikUIExamplesInSwift2.0/TelerikUIExamplesInSwift/AlertViewAnimations.swift
//
//  AlertViewAnimations.swift
//  TelerikUIExamplesInSwift
//
//  Created by Sophia Lazarova on 6/11/15.
//  Copyright (c) 2015 Telerik. All rights reserved.
//

import Foundation

class AlertViewAnimations: ExampleViewController, TKListViewDelegate, TKListViewDataSource {

    var _appearAnimations = ["Fade in", "Scale in", "Slide from top", "Slide from bottom", "Slide from left", "Slide from right"]

    var _hideAnimations = ["Fade out", "Scale out", "Slide to top", "Slide to bottom", "Slide to left", "Slide to right"]
    var _appearAnimationsList: TKListView?
    var _hideAnimationsList: TKListView?
    var _alertView: TKAlert?
    var _appearLabel: UILabel?
    var _hideLabel: UILabel?
    
    override func viewDidLoad() {
        self.setupListView()
        self.view.backgroundColor = UIColor.whiteColor()
       // _alertView = TKAlert(title: "Title", message: "Message", delegate: self, cancelActionTitle: "Cancel", otherActionTitles: "OK")
        _alertView?.style.backgroundStyle = TKAlertBackgroundStyle.Blur
    }
    
    func listView(listView: TKListView!, cellForItemAtIndexPath indexPath: NSIndexPath!) -> TKListViewCell! {
        let cell: TKListViewCell = listView.dequeueReusableCellWithReuseIdentifier("cell", forIndexPath: indexPath) as! TKListViewCell
        
        if(listView == _appearAnimationsList){
            cell.textLabel!.text = _appearAnimations[indexPath.row]
        }
        
        if (listView == _hideAnimationsList) {
            cell.textLabel!.text = _hideAnimations[indexPath.row]
        }
        
        cell.textLabel.font = UIFont.systemFontOfSize(12)
        cell.textLabel.textAlignment = NSTextAlignment.Center
        
        return cell
    }
    
    func listView(listView: TKListView!, numberOfItemsInSection section: Int) -> Int {
          return _appearAnimations.count;
    }
    
    func listView(listView: TKListView!, didSelectItemAtIndexPath indexPath: NSIndexPath!) {
        if(listView == _appearAnimationsList) {
                    switch (indexPath.row) {
                    case 0:
                        _alertView!.style.appearAnimation = TKAlertAnimation.Fade
                        break;
                    case 1:
                        _alertView!.style.appearAnimation = TKAlertAnimation.Scale
                        break;
                    case 2:
                        _alertView!.style.appearAnimation = TKAlertAnimation.SlideFromTop
                        break;
                    case 3:
                        _alertView!.style.appearAnimation = TKAlertAnimation.SlideFromBottom
                        break;
                    case 4:
                        _alertView!.style.appearAnimation = TKAlertAnimation.SlideFromLeft
                        break;
                    case 5:
                        _alertView!.style.appearAnimation = TKAlertAnimation.SlideFromRight
                        break;
                    default:
                        break;
                    }
                }
            
                if(listView == _hideAnimationsList) {
                    switch (indexPath.row) {
                    case 0:
                        _alertView!.style.hideAnimation = TKAlertAnimation.Fade
                        break;
                    case 1:
                        _alertView!.style.hideAnimation = TKAlertAnimation.Scale
                        break;
                    case 2:
                        _alertView!.style.hideAnimation = TKAlertAnimation.SlideFromTop
                        break;
                    case 3:
                        _alertView!.style.hideAnimation = TKAlertAnimation.SlideFromBottom
                        break;
                    case 4:
                        _alertView!.style.hideAnimation = TKAlertAnimation.SlideFromLeft
                        break;
                    case 5:
                        _alertView!.style.hideAnimation = TKAlertAnimation.SlideFromRight
                        break;
                    default:
                        break;
            }
        }
    }
    
    func show(sender: AnyObject)
    {
       _alertView?.show(true)
    }
    
    func setupListView() {
        _appearAnimationsList = TKListView(frame: CGRectMake(0, 26, self.view.frame.size.width/2, self.view.frame.size.height))
        _appearAnimationsList?.dataSource = self
        _appearAnimationsList?.delegate = self
        self.view.addSubview(_appearAnimationsList!)
        _appearAnimationsList?.registerClass(TKListViewCell.self, forCellWithReuseIdentifier: "cell")
        
        _hideAnimationsList = TKListView(frame: CGRectMake(self.view.frame.size.width/2, 90, self.view.frame.size.width/2, self.view.frame.size.height))
        
        _hideAnimationsList?.dataSource = self
        _hideAnimationsList?.delegate = self
        self.view.addSubview(_hideAnimationsList!)
        _hideAnimationsList?.registerClass(TKListViewCell.self, forCellWithReuseIdentifier: "cell")
        
        _appearLabel = UILabel(frame: CGRectMake(0, 70, self.view.frame.size.width/2, 12))
        _appearLabel!.text = "Show animation: "
        _appearLabel?.textColor = UIColor.blackColor()
        _appearLabel?.font = UIFont.systemFontOfSize(12)
        _appearLabel?.textAlignment = NSTextAlignment.Center
        self.view.addSubview(_appearLabel!)
        
        _hideLabel = UILabel(frame: CGRectMake(self.view.frame.size.width/2, 70 , self.view.frame.size.width/2, 12))
        _hideLabel!.text = "Dismiss animation:"
        _hideLabel!.textColor = UIColor.blackColor()
        _hideLabel!.font = UIFont.systemFontOfSize(12)
        _hideLabel!.textAlignment = NSTextAlignment.Center
        self.view.addSubview(_hideLabel!)
        
        let button: UIButton = UIButton(type:UIButtonType.System)
        button.addTarget(self, action: "show:", forControlEvents: UIControlEvents.TouchUpInside)
        button.setTitle("Show Alert", forState: UIControlState.Normal)
        button.backgroundColor = UIColor(red: 0.5, green: 0.7, blue: 0.2, alpha: 0.7)
        button.setTitleColor(UIColor.whiteColor(), forState: UIControlState.Normal)
        button.titleLabel?.font = UIFont(name: "GillSans", size: 18)
        button.frame = CGRectMake((self.view.frame.size.width - 160)/2, 450.0, 160.0, 80.0)
        button.layer.cornerRadius = 80
        button.clipsToBounds = true
        self.view.addSubview(button)

    }
    
}